test/ProgressOnderwijsUtils.Tests/Data/ReadJsonTest.cs

```diff
@@ -0,0 +1,181 @@
+
+                    -- Unicode character strings
+                    , NCharColumn nchar null
+                    , NVarCharColumn nvarchar(32) null
+
+                    -- Binary strings
+                    , BinaryColumn binary(8) null
+
+                    -- Other data types
+                    , UniqueIdentifierColumn uniqueidentifier null
+                    , RowVersionColumn rowversion not null
+                );
+            "
+        ).ExecuteNonQuery(Connection);
+
+        SQL(
+            $@"
+                insert into #ReadJsonTest (
+                    ReadJsonTestId
+                    , BitColumn
+                    , IntColumn
+                    , BigIntColumn
+                    , DecimalColumn
+                    , FloatColumn
+                    , DateColumn
+                    , DateTimeColumn
+                    , DateTime2Column
+                    , CharColumn
+                    , VarCharColumn
+                    , NCharColumn
+                    , NVarCharColumn
+                    , BinaryColumn
+                    , UniqueIdentifierColumn
+                ) values
+                    (1, {true}, {int.MaxValue}, {long.MaxValue}, {0.99m}, {1.234}, {new DateTime(2008, 4, 1)}, {new DateTime(2023, 5, 6, 16, 13, 55)}, {new DateTime(1, 2, 3, 4, 5, 6, 7)}, 'x', 'xyz', N'p', N'pqr', {new byte[] { 1, 2, 3 }}, {"82DBEE37-3AF8-46F2-A403-AE0A1950BC6E"} )
+                    , (2, null, null, null, null, null, null, null, null, null, null, null, null, null, null);
+            "
+        ).ExecuteNonQuery(Connection);
+
+        var pipe = new Pipe();
+        SQL($"select t.* from #ReadJsonTest t").ReadJson(Connection, pipe.Writer, new() { Indented = true, });
+        pipe.Writer.Complete();
+
+        ApprovalTest.CreateHere().AssertUnchangedAndSave(Encoding.UTF8.GetString(pipe.Reader.ReadAsync().GetAwaiter().GetResult().Buffer));
+    }
+
+    enum ReadJsonPocoTestId { }
+
+    sealed record ReadJsonPocoTest : IWrittenImplicitly
+    {
+        public ReadJsonPocoTestId ReadJsonPocoTestId { get; init; }
+        public bool BooleanColumn { get; init; }
+        public int? NumberColumn { get; init; }
+        public string? StringColumn { get; init; }
+        public DateTime? DateTimeColumn { get; init; }
+        public byte[]? BinaryColumn { get; init; }
+        public required byte[] RowVersionColumn { get; init; }
+    }
+
+    [Fact]
+    public void Deserialize_ReadJson_gives_the_same_result_as_ReadPocos()
+    {
+        SQL(
+            $@"
+                create table #ReadJsonPocoTest (
+                    ReadJsonPocoTestId int not null
+                    , BooleanColumn bit not null
+                    , NumberColumn int null
+                    , StringColumn nvarchar(32) null
+                    , DateTimeColumn datetime2 null
+                    , BinaryColumn varbinary(32) null
+                    , RowVersionColumn rowversion not null
```

Comment:
missing bigint, decimal, +float.

(Short term workaround if those aren't workable: crash instead of generating un-parseable json)